src/pygeoprocessing/geoprocessing.py

```diff
@@ -1034,47 +1034,97 @@ def create_raster_from_vector_extents(
+        target_srs_wkt (string): The SRS of the target raster, in Well-Known
+            Text format.
+        target_nodata (float): The nodata value of the target raster, or
+            ``None`` if no nodata value is to be set.
+        fill_value=None (number): If provided, the value that the target raster
+            should be filled with.
+        raster_driver_creation_tuple (tuple): a tuple containing a GDAL driver
+            name string as the first element and a GDAL creation options
+            tuple/list as the second. Defaults to a GTiff driver tuple
+            defined at geoprocessing.DEFAULT_GTIFF_CREATION_TUPLE_OPTIONS.
+
+    Returns:
+        ``None``
+    """
+    if target_pixel_type not in _VALID_GDAL_TYPES:
+        raise ValueError(
+            f'Invalid target type, should be a gdal.GDT_* type, received '
+            f'"{target_pixel_type}"')
+
+    bbox_minx, bbox_miny, bbox_maxx, bbox_maxy = target_bounding_box
+
+    driver = gdal.GetDriverByName(raster_driver_creation_tuple[0])
+    n_bands = 1
     n_cols = int(numpy.ceil(
-        abs((shp_extent[1] - shp_extent[0]) / target_pixel_size[0])))
+        abs((bbox_maxx - bbox_minx) / target_pixel_size[0])))
     n_cols = max(1, n_cols)
 
     n_rows = int(numpy.ceil(
-        abs((shp_extent[3] - shp_extent[2]) / target_pixel_size[1])))
+        abs((bbox_maxy - bbox_miny) / target_pixel_size[1])))
     n_rows = max(1, n_rows)
 
-    driver = gdal.GetDriverByName(raster_driver_creation_tuple[0])
-    n_bands = 1
     raster = driver.Create(
         target_raster_path, n_cols, n_rows, n_bands, target_pixel_type,
         options=raster_driver_creation_tuple[1])
-    raster.GetRasterBand(1).SetNoDataValue(target_nodata)
+    raster.SetProjection(target_srs_wkt)
 
     # Set the transform based on the upper left corner and given pixel
-    # dimensions
+    # dimensions.  Bounding box is in format [minx, miny, maxx, maxy]
     if target_pixel_size[0] < 0:
-        x_source = shp_extent[1]
+        x_source = bbox_maxx
     else:
-        x_source = shp_extent[0]
+        x_source = bbox_minx
     if target_pixel_size[1] < 0:
-        y_source = shp_extent[3]
+        y_source = bbox_maxy
     else:
-        y_source = shp_extent[2]
+        y_source = bbox_miny
     raster_transform = [
         x_source, target_pixel_size[0], 0.0,
         y_source, 0.0, target_pixel_size[1]]
     raster.SetGeoTransform(raster_transform)
 
-    # Use the same projection on the raster as the shapefile
-    raster.SetProjection(vector.GetLayer(0).GetSpatialRef().ExportToWkt())
-
-    # Initialize everything to nodata
+    # Fill the band if requested.
+    band = raster.GetRasterBand(1)
     if fill_value is not None:
-        band = raster.GetRasterBand(1)
         band.Fill(fill_value)
-        band = None
-    vector = None
+
+    # Set the nodata value.
+    if target_nodata is not None:
+        band.SetNoDataValue(float(target_nodata))
```

Comment:
I'm curious why casting to float here?

Reply:
Yeah, this is just for safety, in case the user provides a nodata value with an interesting type, like any of the numpy dtypes, `decimal.Decimal`, etc.  If anything other than an `int` or `float` is provided, there'll be a `TypeError`.  Casting to a `float` will just ensure that the nodata value will be of a type that GDAL can handle.